src/theme/index.tsx

```diff
@@ -0,0 +1,22 @@
+import { CssBaseline, createTheme, PaletteMode } from '@mui/material';
+import { ThemeProvider } from '@mui/material/styles';
+import { useMemo } from 'react';
+import { getDesignTokens } from './themeTokens';
+import { useAppSelector } from '../hooks/redux-hooks';
+import { getProfile } from '../stores/slices/authSlice';
+
+const ThemeWrapper = ({ children }: { children: React.ReactNode }) => {
+  const uiTheme = useAppSelector(getProfile)?.uiTheme || localStorage.getItem('ui_theme');
+  const mode: PaletteMode = uiTheme === 'DARK' ? 'dark' : 'light';
```

Comment:
Here is a little constants mess. You can create a function that converts `ThemeMode` to `PalletteMode`. It will help to avoid inline conditions and use constants outside that function.